src/Storages/MergeTree/MergeTreeDataPartWriterOnDisk.cpp

```diff
@@ -152,11 +154,13 @@ static size_t computeIndexGranularityImpl(
             index_granularity_for_block = index_granularity_bytes / size_of_row_in_bytes;
         }
     }
-    if (index_granularity_for_block == 0) /// very rare case when index granularity bytes less then single row
-        index_granularity_for_block = 1;
-
     /// We should be less or equal than fixed index granularity
     index_granularity_for_block = std::min(fixed_index_granularity_rows, index_granularity_for_block);
+
```

Comment:
Why you changed the order or these lines?

Reply:
It will saturate 0 to 1 even if `fixed_index_granularity_rows` was 0 in this function.